pkg/auth/ucp/kubeconfig.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package ucp provides UCP authentication functions.
+package ucp
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	kubeutils "github.com/vmware-tanzu/tanzu-cli/pkg/auth/utils/kubeconfig"
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
+)
+
+func GetUCPKubeconfig(c *configtypes.Context, endpoint, orgID, endpointCACertPath string, skipTLSVerify bool) (string, string, string, error) {
+	clusterAPIServerURL := strings.TrimSpace(endpoint)
+	if !strings.HasPrefix(clusterAPIServerURL, "https://") && !strings.HasPrefix(clusterAPIServerURL, "http://") {
+		clusterAPIServerURL = "https://" + clusterAPIServerURL
+	}
+	clusterAPIServerURL = clusterAPIServerURL + "/org/" + orgID
+
+	clusterCACertData := ""
+	if endpointCACertPath != "" {
+		fileBytes, err := os.ReadFile(endpointCACertPath)
+		if err != nil {
+			return "", "", "", errors.Wrapf(err, "error reading CA certificate file %s", endpointCACertPath)
+		}
+		clusterCACertData = base64.StdEncoding.EncodeToString(fileBytes)
+	}
+
+	contextName := "tanzu-cli-" + c.Name
```

Comment:
Nit: might be good to write some tiny helper functions to construct these, and use them where appropriate.